### PR TITLE
How-to: Assign IPs based on topology

### PR DESCRIPTION
**Pull request by bmckercher123**
Converted existing doc to How to format.

- Original author, Casey
- Can merge at any time

### CODE REVIEW

**Comment by caseydavenport** [comment]
Deploy preview for *calico* ready!

Built with commit a12ef18ff7a6ce3ad5c9ec51517a534a6a48524b

https://deploy-preview-2907--calico.netlify.com

File: master/networking/assign-ip-addresses-topology.md [hunk 1]
```diff
@@ -1,47 +1,68 @@
...
+
+Node selection-based IP address assignment is exactly what it sounds like: node labels are set, and then the appropriate node selectors on the desired IP pool resources are set. 
+
+#### Best practice
+
+Nodes only assign workload addresses from IP pools which select them. To avoid having a workload not get an IP and fail to start, it is important to ensure that all nodes are selected by at least one IP pool.
+
+### Before you begin
+
+**Required**: etcd datatstore
```
Comment (caseydavenport) [inline master/networking/assign-ip-addresses-topology.md]: This isn't true any more - this feature is supported for both data stores.

File: master/networking/assign-ip-addresses-topology.md [hunk 2]
```diff
@@ -1,47 +1,68 @@
...
-In this example, we created a cluster with four nodes across two racks
-(two nodes/rack). Consider the following:
+#### IP address assignment
+
+Topology-based IP address assignment requires addresses to be per-host (node). 
+As such, Kubernetes annotations cannot be used because annotations are only per-namespace and per-pod. And although you can configure IP addresses for nodes in the CNI configuration, you are making changes within the host’s file system. The best option is to use node-selection IP address assignment using IP pools.
+
+#### Node-selection IP address management
+
+Node selection-based IP address assignment is exactly what it sounds like: node labels are set, and then the appropriate node selectors on the desired IP pool resources are set.
```
Comment (caseydavenport) [inline master/networking/assign-ip-addresses-topology.md]: This sentence doesn't really describe what node-selection is.

I might say "node labels are set, and Calico uses node selectors to decide which IP pools to use when assigning IP addresses to that node"

File: master/networking/assign-ip-addresses-topology.md (hunk 1 above)
Comment (bmckercher123) [reply master/networking/assign-ip-addresses-topology.md]: Fixed.

File: master/networking/assign-ip-addresses-topology.md (hunk 2 above)
Comment (bmckercher123) [reply master/networking/assign-ip-addresses-topology.md]: Fixed.

**Comment by caseydavenport** [comment]
@bmckercher123 looks like there are conflicts in this PR now and it needs a rebase.

**Pull request merged**